Adding Living Style Guide (as part of Rails project)

myabc opened the pull request:
https://www.openproject.org/work_packages/7630

I'm not a fan of monolithic applications, but for now at least, it's a lot less complex to keep the style guide in the same repository and use the built-in LSG Rails integration. Replaces #1280.

Style guide is available at `localhost:3000/assets/styleguide.html`

myabc committed:
Set up Living Style Guide; add first documentation

myabc committed:
Add livingstyleguide gem, its Compass dependency

Only pre-release versions of Compass are currently Sass 3.3
compatible.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Normalise naming of all Sass partials

A Sass source should never be compiled to a separate CSS file if its
filename:
  - begins with underscore.
  - has a .sass extension (and only .sass).

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add simplified 'meta' stylesheet for LSG

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Use sprockets backport-compatible sass-rails fork

With 3.2.x sass-rails release, we hit NoMethodErrors with the Living
Style Guide attempting to handle asset helpers:

    undefined method `image_path' for nil:NilClass

This resolves the issue, for now, but moving to Rails 4 is the better
medium-term solution.

See also https://discussion.heroku.com/t/using-the-rails-4-asset-pipeline-in-rails-3-apps-for-faster-deploys/205

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove Sprockets::Helpers::RailsHelper usage

This Module is no longer present in sprockets-rails 2.0.0.backport1.
See previous commit c0dc50e9.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Update i18n-js, no longer specifying single ref

Signed-off-by: Alex Coles <alex@alexbcoles.com>

tilt commented:
heroic

tessi commented on Gemfile:
```diff
@@ -107,11 +107,14 @@ group :production do
 
 group :assets do
-  gem 'sass-rails',   '~> 3.2.3'
+  gem 'sass-rails', git: 'https://github.com/guilleiguaran/sass-rails.git', branch: 'backport'
```
I'd love to see a comment on why we have to use he git repo (instead of a version from rubygems here)

myabc replied:
@tessi in addition to the commit message? (c0dc50e9). Our `Gemfile` is already rather cluttered IMO.

I _do_ have reservations about depending on relatively _obscure_ forks. Perhaps a better strategy for us would be to fork this fork into the finnlabs account?

These niggling asset problems won't go away until we upgrade to Rails 4.0.

tessi replied:
Ah, haven't seen the commit message. I'm fine with that :)

myabc committed:
Namespace Hooks module to avoid naming collision

Our own top-level Hooks module collides with that provided in the
hooks gem, now an indirect dependency (needed by livingstyleguide).

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Re-include asset helpers in ViewListener

In sprockets-rails 2.0.0.backport1, `Sprockets::Helpers::RailsHelper`
now appears to be called `Sprockets::Rails::Helper`.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove Gemfile assets group

Recommendation in the following article is to remove this group when
using backported asset pipeline-related gems:
https://discussion.heroku.com/t/using-the-rails-4-asset-pipeline-in-rails-3-apps-for-faster-deploys/205

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix require of renamed OpenProject::Hooks module

See ee70ffbc.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Attempt workaround for failing theme asset specs

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Attempt workaround for LSG asset loading

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Living Style Guide: Document fonts

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add compass-rails (master)

This appears to fix issues using asset paths helpers with the Living
Style Guide. See https://github.com/hagenburger/livingstyleguide/issues/70

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Revert "Attempt workaround for LSG asset loading"

Adding compass-rails (previous commit 9689b158) solves the same issue.

This reverts commit 67e56122153a846a078118683c8e8195de1b5aa4.

myabc commented:
I dislike the hack in b0f758fc, but otherwise this should be mergeable.

Commit:
Add styles for work package table

- added styles for work packages tables
